src/Shared/NativeMethodsShared.cs

```diff
@@ -1181,17 +1235,76 @@ internal static int GetParentProcessId(int processId)
         /// Internal, optimized GetCurrentDirectory implementation that simply delegates to the native method
         /// </summary>
         /// <returns></returns>
-        internal static string GetCurrentDirectory()
+        internal unsafe static string GetCurrentDirectory()
         {
+#if FEATURE_LEGACY_GETCURRENTDIRECTORY
```

Comment:
What is the meaning of FEATURE_LEGACY_GETCURRENTDIRECTORY? I see that it does not get set.

Without it being set, this method will end up calling the managed implementation on Windows, which will regress perf.

Reply:
Missed adding it to this PR, it should only get set under net46